include/gnc/constants.hpp

```diff
@@ -41,11 +45,17 @@ GNC_TRACKED_CONSTANT(constexpr static lin::Vector4d, q_eci_perifocal, 0.12745663
 
 GNC_TRACKED_CONSTANT(constexpr static double, b_noise_floor, 0.0);
 
-GNC_TRACKED_CONSTANT(constexpr static double, max_mtr_moment, 0.113337 / 2.0);
+GNC_TRACKED_CONSTANT(constexpr static float, max_mtr_moment, 0.113337f / 2.0f);
+
+extern lin::Matrix3x3f J_sat;
```

Comment:
If these are supposed to actually change in normal mission ops, then they should be part of the estimator or controller interface and state (and probably the MATLAB state and interface) instead of just C++ global variables that are "constant" but not really.

Reply:
I feel like having the controller and estimator state and interface in global variables will just get messy and make testing, debugging, and documentation a lot harder. One quick fix is to just create a big class called Context that manages all of these shared parameters, or they could just go into the state field registry (maybe have a control task that manages the shared gnc parameters?). But ideally if the parameters aren't really shared by lots of different things they should be part of the specific controller or estimators internal state and interface.

Reply:
Here I was just following the paradigm we have set in flight software. We have a large set of values that we call "constants" that normally live in state fields and are sometimes editable from the ground. In the case of the controllers, I just pushed any "input" that didn't change to frequently out into global constants so we don't have to recopy the values on each call to the controller each time. Furthermore, something like the moment of inertia will definitely be used in the attitude estimator as well except we have no reason for them to be working off of different values.

Reply:
I'd take a look at the issue linked above from FSW talking about wrapping these editable constants with state fields in flight software. Obviously, it'd be best to make them state fields anyway but we are in different repos so here we are haha And lastly, we could make a context class but then we're never going to have more than one so we could just throw it in a namespace no?

Reply:
@nhz2 since we're not planning to modify these things under normal operations I don't think it makes sense to pass them all to the controller on every function call. It's simpler to have them defined as constants and it makes sense to retain the ability to edit them in some off-nominal case if we need to.

Reply:
If we don't want to worry about the docked case and rely on edits from the ground to change the controller state then I agree with that. Our current mission plan just turns off ADCS when we dock and wait for the ground anyway haha I misspoke in the description above then.

Reply:
Ok if they are only getting changed from ground in some kind of debug situation thats fine, just realize that it will be very very hard and annoying to test changing any C++ global variables in a MATLAB sim because they will be shared by the entire MATLAB context. So if you are fine not testing the behavior of changing those parameters during a MATLAB sim, and forcing both leader and follower to have the same values, then they can stay as globals.